data_import/models.py

```diff
@@ -224,3 +230,41 @@ class TestUserData(models.Model):
         related_name="test_user_data",
         on_delete=models.CASCADE,
     )
+
+
+class DataType(models.Model):
+    """
+    Describes the types of data a DataFile can contain.
+    """
+
+    name = models.CharField(
+        max_length=128, blank=False, unique=True, validators=[charvalidator]
+    )
+    parent = models.ForeignKey("self", blank=True, null=True, on_delete=models.CASCADE)
```

Comment:
here's a good spot to use `related_name=children` so one can do `parent_datatype.children.all()`, otherwise it's `parent_datatype.datatype_set.all()` which isn't as intuitive.

also I think `on_delete=models.CASCADE` is a really bad idea for DataTypes. I would go with `PROTECT` to prevent orphaned DataTypes in the tree.